en/guide/message_signing.md

```diff
@@ -1,121 +1,157 @@
 
-The purpose of the `link_id` field in the *MAVLink 2* signing structure is to prevent cross-channel replay attacks. Without the `link_id` an attacker could record a packet (such as a disarm request) on one channel, then play it back on a different channel.
 
-The intention with the link IDs is that each channel of communication between an autopilot and a GCS uses a different link ID. There is no requirement that the same link ID be used in both directions however.
+## Accepting Signed Packets
 
-For C implementations the obvious mechanism is to use the MAVLink channel number as the link ID. That works well for an autopilot, but runs into an issue for a GCS implementation. The issue is that a user may launch multiple GCS instances talking to the same autopilot via different communication links (such as two radios, or USB and a radio). These multiple GCS instances will not be aware of each other, and so may choose the same link ID. If that happens then a large number of correctly signed packets will be rejected by the autopilot as they will have timestamps that are older than the timestamp received for the same stream tuple on the other communication link.
+When a signed packet arrives it should be discarded if the:
+* Timestamp is older than the previous packet from the same logical stream - where a logical stream is defined as the sequence of MAVLink packets with the same (`SystemID`, `ComponentID`, `LinkID`) tuple.
+* Computed 48 bit signature does not match the signature included in the packet. 
+* The timestamp is more than 1 minute (6,000,000) behind the local system’s timestamp.
 
-The solution adopted for MAVProxy is shown below:
 
-```
-if (msg.get_signed() and
-	self.mav.signing.link_id == 0 and
-	msg.get_link_id() != 0 and
-	self.target_system == msg.get_srcSystem() and
-	self.target_component == msg.get_srcComponent()):
-	# change to link_id from incoming packet
-	self.mav.signing.link_id = msg.get_link_id()
-```
+## Accepting Unsigned Packets {#accepting_unsigned_packets}
+
+MAVLink libraries should provide a mechanism that allows a system to conditionally accept *unsigned* packets.
+
+The rules for accepting these packets will be implementation specific, but could be based on a combination of a parameter setting, transport type, message type, (in)compatibility flags etc.
+
+A few rules that may be useful for some systems are:
+* All unsigned packets accepted based on a system-specific parameter.
+* All unsigned packets accepted if the connection is over a "secure channel" (e.g. local USB cable or local wired Ethernet cable).
+* `RADIO_STATUS` packets are always accepted without signing (to make life easier for telemetry radios).
+* All unsigned packets accepted when in an "unsigned mode" (perhaps triggered by a hardware button pressed on boot).
+* All other packets will be rejected if not signed!
+
+
+## Accepting Incorrectly Signed Packets {#accepting_incorrectly_signed_packets}
```

Comment:
It's probably obvious but this deserves a warning box that points out that you should *never* use an incorrectly signed packets timestamp to update the systems idea of signing timestamps. If you accept timestamps from this you can be introduced to all sorts of fun behaviors by malicious users.

Reply:
Agreed. However I put this in the Timestamp section and linked to here.